include/envoy/event/dispatcher.h

```diff
@@ -197,6 +197,8 @@ class Dispatcher : public DispatcherBase {
                          Network::TransportSocketPtr&& transport_socket,
                          const Network::ConnectionSocket::OptionsSharedPtr& options) PURE;
 
+  virtual Network::ConnectionFactory& connectionFactory() PURE;
```

Comment:
As I promised, 
I extract the  responsiblity of createServerConnection and createServerConnection to ConnectionFactory.

For dementration createClientConnection() calls connectionFactory(). Will remove createClientConnection and createServerConnection from the dispatcher interface.

So whatever internal listener/connection changes won't touch dispatcher